File: software/backend/src/main/java/ru/scms/back/dto/CaseDto.java
package ru.scms.back.dto;

import java.time.LocalDateTime;
import ru.scms.back.enums.CaseStatus;
import ru.scms.back.enums.IncidentType;
import ru.scms.back.enums.SleeveGender;

public record CaseDto(
        Long id,
        String code,
        Long orderId,
        Long methUserId,
        String methUserName,
        Long sleeveId,
        String sleeveCode,
        SleeveGender sleeveGender,
        Integer sleeveHeight,
        Long stackId,
        String stackCode,
        CaseStatus status,
        String needlecasterName,
        LocalDateTime startTime,
        LocalDateTime endTime,
        String result,
        IncidentType incidentType,
        String incidentNote,
        LocalDateTime createdAt) {}

File: software/client-portal/src/shared/api/dto/caseDto.ts
import type { CaseStatus, IncidentType, SleeveGender } from "./enums";

export interface CaseDto {
  id: number;
  code: string;
  orderId: number | null;
  methUserId: number;
  methUserName: string | null;
  sleeveId: number | null;
  sleeveCode: string | null;
  sleeveGender: SleeveGender | null;
  sleeveHeight: number | null;
  stackId: number | null;
  stackCode: string | null;
  status: CaseStatus;
  needlecasterName: string | null;
  startTime: string | null;
  endTime: string | null;
  result: string | null;
  incidentType: IncidentType | null;
  incidentNote: string | null;
  createdAt: string;
}
